sheet_to_triples/__main__.py

```diff
@@ -27,12 +28,13 @@ def _is_book_path(path):
 
 
 def _parse_book_path(path):
-    if os.path.isdir(path):
+    st = os.stat(path)
+    if stat.S_ISDIR(st.st_mode):
         for root, dirs, files in os.walk(path):
             for filepath in filter(_is_book_path, files):
                 yield os.path.join(root, filepath)
             return
-    else:
+    elif stat.S_ISREG(st.st_mode):
```

Comment:
What's wrong with `os.path.isfile`?

Reply:
Two stat calls! The waste! :)

I'd prefer if there was better spelling here that made it clear we're interested in the properties of one file, and are putting the behaviour into three buckets - the stat version here is unfortunately rather ugly.

Reply:
I am very surprised that `list(os.walk('nosuch')) == []` rather than throwing, which would seem like the right response.